### PR TITLE
[WFLY-9859] Add client compatibility tests using the 3.0.x client

### PR DESCRIPTION
**Pull request by bstansberry**
https://issues.jboss.org/browse/WFLY-9859

Updates the standard test proving that legacy clients can perform basic management ops against the current server.

It's not critical to me that this gets merged, as we are considering dropping this test class as redundant (see WFLY-9264). It was just 10 mins work to add it as I did https://issues.jboss.org/browse/WFCORE-3391 so I figured why not do it and confirm it passes. :)

### CODE REVIEW

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/management/api/ClientCompatibilityUnitTestCase.java [hunk 1]
```diff
@@ -295,7 +305,6 @@ protected static ModelControllerClient createClient(final String artifact, final
 
         final ChildFirstClassLoaderBuilder classLoaderBuilder = new ChildFirstClassLoaderBuilder(false);
         classLoaderBuilder.addRecursiveMavenResourceURL(artifact + ":" + version, excludes);
-        classLoaderBuilder.addParentFirstClassPattern("org.jboss.as.controller.client.ModelControllerClientConfiguration");
```
Comment (rsvoboda) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/management/api/ClientCompatibilityUnitTestCase.java]: Why is this removed ?

Comment (bstansberry) [reply testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/management/api/ClientCompatibilityUnitTestCase.java]: @rsvoboda Because its presence was wrong and results in the test not working. :)  ModelControllerClientConfiguration is not used directly by the test driver and therefore doesn't should not be on the parent-first class path. If it is on the parent first classpath, the legacy client being constructed instead of using the ModelControllerClientConfiguration class that shipped with the legacy driver instead uses the one that's in current WF Core. That results in an incorrectly constructed client that fails.

Comment (rsvoboda) [reply testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/management/api/ClientCompatibilityUnitTestCase.java]: Thanks for sharing details
